middleware/classes.py

```diff
@@ -150,3 +154,52 @@ def process_exception(self, request, exception):
             response = None
 
         return response
+
+
+class MobileDetectMiddleware:
+    """Mobile Detect Middleware
+
+    Redirects mobile devices to `/mobile` path if `early_access_code` is in the
+    URL. The Deep Linking will catch the URL and it will be opened in the app.
+
+    If the user does not have the app on their device, the `/mobile` path
+    should have a QR code and link to the app store.
+    """
+
+    def __init__(self, get_response=None):
+        if get_response is not None:
+            self.get_response = get_response
+
+    def __call__(self, request):
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.
+        request = self.process_request(request)
+        response = self.get_response(request)
+
+        if FEATURE_EARLY_ACCESS.is_enabled:
+            response = self.redirect_mobile_early_access(request, response)
+        else:
+            pass
+
+        return response
+
+    def process_request(self, request):
+        user_agent = parse(request.META.get('HTTP_USER_AGENT', ''))
+        request.user_agent = user_agent
+        return request
+
+    def redirect_mobile_early_access(self, request, response):
+        is_mobile_path = request.path.startswith('/mobile')
```

Comment:
The `/mobile` should be a HTK setting so that it can be changed depending on the project needs.